src/robot/libdocpkg/specbuilder.py

```diff
@@ -90,3 +93,32 @@ def _create_arguments(self, elem):
                 spec.types = {}
             spec.types[name] = tuple(t.text for t in type_elems)
         return spec
+
+    def _create_data_types(self, spec):
+        enums = [self._create_enum_doc(dt)
+                 for dt in spec.findall('data_types/enums/enum')]
+        typed_dicts = [self._create_typed_dict_doc(dt)
+                       for dt in spec.findall('data_types/typed_dicts/typed_dict')]
+        return enums + typed_dicts
+
+    def _create_enum_doc(self, dt):
+        return EnumDoc(name=dt.get('name'),
+                       type='Enum',
+                       doc=dt.find('doc').text or '',
+                       members=[{'name': member.get('name'),
+                                 'value': member.get('value')}
+                                for member in dt.findall('members/member')])
+
+    def _create_typed_dict_doc(self, dt):
+        items = []
+        for item in dt.findall('items/item'):
+            required = item.get('required', None)
+            if required is not None:
+                required = True if required == 'true' else False
+            items.append({'key': item.get('key'),
+                          'type': item.get('type'),
+                          'required': required})
+        return TypedDictDoc(name=dt.get('name'),
+                            type='TypedDict',
```

Comment:
Needing to pass `TypedDict` as type explicitly is somewhat strange. Couldn't that be de default with `TypedDictDoct`?